### ISSUE TITLE
Source Control not working using Remote SSH

### ISSUE THREAD

**Issue by tttapa** [microsoft/vscode-remote-release#1848]
Issue Type: <b>Bug</b>

When connecting to another computer over SSH, the Source Control tab shows "No source control providers registered", even though the workspace folder I have open contains a Git repository.  
This problem arose after updating to VS Code version 1.40.0, it worked fine in the previous version.

1. Ctrl+Shift+P > Remote-SSH: Connect to Host...
2. Select SSH host
3. Open Folder
4. Select a folder containing a Git repository
5. Open the Source Control tab (Ctrl+Shift+G)

![vscode-git- not-working-ssh](https://user-images.githubusercontent.com/21296729/68528175-a3658480-02ef-11ea-98e4-499ee152c4db.png)

When opening a local workspace (without Remote - SSH), source control seems to work fine.

---

VS Code version: Code 1.40.0 (86405ea23e3937316009fc27c9361deee66ffbf5, 2019-11-06T18:14:08.920Z)
OS version: Linux x64 4.15.0-66-generic
Remote OS version: Linux x64 4.15.0-66-generic
<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 3428)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|15.39GB (5.84GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: cs|
|OS|Linux x64 4.15.0-66-generic|
|CPUs|Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz (4 x 3313)|
|Memory (System)|15.55GB (2.19GB free)|
|VM|0%|

</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-containers|ms-|0.83.0
remote-ssh|ms-|0.47.2
remote-ssh-edit|ms-|0.47.2
remote-wsl|ms-|0.40.2
vscode-remote-extensionpack|ms-|0.17.0
material-icon-theme|PKi|3.9.1
fortranbreaker|eki|0.0.4
fortran|Gim|0.2.0
fortran-ls|han|0.6.0
linter-gfortran|krv|2.1.1
python|ms-|2019.10.44104
cpptools|ms-|0.26.1
cmake|twx|0.0.17

</details>

**Comment by sandy081** [comment]
@roblourens @alexandrudima passing it over to you

**Comment by roblourens** [comment]
You can try running the Kill VS Code Server command to install from scratch

**Comment by tttapa** [comment]
*TL;DR*: The Kill VS Code Server command + manually deleting a folder, then re-installing seems to have fixed the remote extension problems.

*Long story*:

The Kill VS Code Server command didn't work for me. I tried closing VSCode (I only had local folders open), and then running it again multiple times, but it produced the same error message over and over again:

```
[18:29:24.331] remote-ssh@0.47.2
[18:29:24.331] linux x64
[18:29:24.441] Picking SSH host
[18:29:29.368] Selected cs
[18:29:29.371] Running script on remote: 
		kill -9 `ps ax | grep "out/vs/server/main.js" | grep -v grep | awk '{print $1}'`
		kill -9 `ps ax | grep "remoteExtensionHostAgent.js" | grep -v grep | awk '{print $1}'`
		kill -9 `ps ax | grep "watcherService" | grep -v grep | awk '{print $1}'`
		rm -rf $HOME/.vscode-server/bin
		rm -rf $HOME/.vscode-remote/bin
	
[18:29:29.375] Testing ssh with ssh -V
[18:29:29.404] ssh exited with code: 0
[18:29:29.404] Got stderr from ssh: OpenSSH_7.2p2 Ubuntu-4ubuntu2.8, OpenSSL 1.0.2g  1 Mar 2016
[18:29:29.407] Running script with connection command: ssh -T -o ClearAllForwardings=true -o ConnectTimeout=15 cs bash
[18:29:30.874] > kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
[18:29:30.928] > kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
> kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
> rm: cannot remove '/home/r0663636/.vscode-server/bin/86405ea23e3937316009fc27c9361deee66ffbf5/.nfs00000000023a3cf30000027d': Device or resource busy
[18:29:31.181] "Kill VS Code Server" terminal command done
```
Now I couldn't connect to the remote anymore, it just kept failing and asking me to retry.

I then just opened an SSH shell and tried to manually delete the busy file, but it didn't exist, so I just deleted the empty folder it was in (`rmdir /home/r0663636/.vscode-server/bin/86405ea23e3937316009fc27c9361deee66ffbf5`), and ran the Kill VS Code Server command again.  
That seemed to work, and I can now connect again.  
All extensions (I tested Git and JSON syntax highlighting) seem to work again as well.

As far as I'm concerned, you can close this issue, as it looks like a one-off thing.

Thanks to everyone who commented!

**Comment by roblourens** [comment]
Oh interesting, that file is from your NFS system, https://serverfault.com/questions/201294/nfsxxxx-files-appearing-what-are-those

I guess something kept it open. Glad you figured it out

**Comment by roblourens** [comment]
Oh interesting, that file is from your NFS system, https://serverfault.com/questions/201294/nfsxxxx-files-appearing-what-are-those

I guess something kept it open. Glad you got it worked out
